vignettes/vignette_best_practices.Rmd

```diff
@@ -0,0 +1,151 @@
+---
+title: "Best practices when writing vignette in admiral"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Best practices when writing vignette in admiral}
+  %\VignetteEngine{knitr::rmarkdown}
+---
+
+```{r setup0, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+# Introduction
+
+This guidance ensures consistency across the vignettes in the admiral package in terms of content, structure and code execution. 
+As a general rule, the vignette workflow defined in [r-pkgs.org](https://r-pkgs.org/vignettes.html) is followed.
+
+
+# Metadata
+
+Each vignette in admiral should start with the following metadata.
+
+```
+---
+title: "<Title>"
+output:  rmarkdown::html_vignette:
+vignette: >
+  %\VignetteIndexEntry{<Title>}
+  %\VignetteEngine{knitr::rmarkdown}
+---
+```
+
+The `<Title>` of the vignette should be meaningful. 
+
+# Markdown
+
+## Default options
+
+    `r ''````{r setup, include=FALSE}
+    knitr::opts_chunk$set(
+      collapse = TRUE,
+      comment = "#>"
+      )
+    ```
+
+## Required Packages
+
+    `r ''````{r, warning=FALSE, message=FALSE}
+    library(admiral)
+    #<all packages used in the vignette>
+    ```
+The `warning = FALSE` and `message = FALSE` options are there to prevent the usual messages:
+
+<span style="color: red;font-family: textmate; font-size = 12pt;" >Attaching package: 'xxxx'</span>
+
+<span style="color: red;font-family: textmate; font-size = 12pt;">The following objects are masked from 'package:yyyyy':</span>
+
+<span style="color: red;font-family: textmate; font-size = 12pt;">  fun1, fun2</span>
+
+
+
+## Format Sections
+
+### Table of contents
+
+Headings must start from Heading 1:
+
+<span style="color: red;" > **Note: some vignettes starts the heading at H2** </span>
+
+```
+# Heading 1
+## Heading 2
+
+```
+
+The first section (# Introduction) gives a brief introduction.
+
+For vignettes describing an ADaM workflow, the second section will summarize the programming workflow.
```

Comment:
Would it make sense to separate general rules and conventions and rules and convention for workflow vignettes?

Reply:
done, please review and suggest if needed